tests-e2e/core/promotionrun_controller_test.go

```diff
@@ -143,6 +147,118 @@ var _ = Describe("Application Promotion Run E2E Tests.", func() {
 			}
 
 			Eventually(promotionRun, "3m", "1s").Should(promotionRunFixture.HaveStatusComplete(expectedPromotionRunStatus))
+
+			envVariable := os.Getenv("DISABLE_APPSTUDIO_WEBHOOK")
+			// Don't run webhook checks if it is disabled.
+			if envVariable != "" && !strings.EqualFold(envVariable, "true") {
+				By("Validate CR Webhooks.")
```

Comment:
Would it be possible to move the webhook tests into their own file, e.g. `webhook_tests.go`, something like that? In my earlier PR comment, I was confused as I had assumed that the tests in `promotionrun_controller_test.go` were only related to the promotionrun CR. But as you mentioned, this block of tests is testing webhooks for all components.